core/src/main/java/software/amazon/awssdk/http/SdkHttpFullRequestAdapter.java

```diff
@@ -35,18 +36,20 @@ public static SdkHttpFullRequest toHttpFullRequest(Request<?> request) {
 
     public static SdkHttpFullRequest.Builder toMutableHttpFullRequest(Request<?> request) {
         return SdkHttpFullRequest.builder()
-                                 .content(request.getContent())
-                                 .httpMethod(SdkHttpMethod.fromValue(request.getHttpMethod().name()))
+                                 .protocol(request.getEndpoint().getScheme())
```

Comment:
The reason I was a bit hesitant to do that is because a URI has more than just a scheme, host and port. We'd probably want to decode the path, query parameters, etc. That felt like overkill in our case.